### PR TITLE
Feature/37

### PR DESCRIPTION
**Pull request by uedamiku**
注文時にメールを送る機能を実装

### CODE REVIEW

File: src/main/java/com/example/service/SendOrderService.java [hunk 1]
```diff
@@ -9,8 +8,10 @@
 
 import org.springframework.beans.BeanUtils;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mail.MailException;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
+import org.springframework.mail.SimpleMailMessage;
```
Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:14]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'org.springframework.mail.SimpleMailMessage' import. Should be before 'org.springframework.transaction.annotation.Transactional'.

File: src/main/java/com/example/service/SendOrderService.java [hunk 2]
```diff
@@ -21,6 +22,8 @@
 import com.example.repository.ItemRepository;
 import com.example.repository.OrderRepository;
 import com.example.repository.ToppingRepository;
+import org.springframework.mail.javamail.JavaMailSender;
```
Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:25]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'org.springframework.mail.javamail.JavaMailSender' import. Should be before 'org.springframework.transaction.annotation.Transactional'.

File: src/main/java/com/example/service/SendOrderService.java [hunk 3]
```diff
@@ -21,6 +22,8 @@
...
+import org.springframework.scheduling.annotation.Async;
```
Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:26]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'org.springframework.scheduling.annotation.Async' import. Should be before 'org.springframework.transaction.annotation.Transactional'.

File: src/main/java/com/example/service/SendOrderService.java [hunk 4]
```diff
@@ -33,41 +36,45 @@
...
 	@Autowired
 	private OrderRepository orderRepository;
-	
+
 	@Autowired
 	private ItemRepository itemRepository;
 
 	@Autowired
 	private ToppingRepository toppingRepository;
+
```
Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:45]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/service/SendOrderService.java [hunk 5]
```diff
@@ -33,41 +36,45 @@
...
 	private OrderRepository orderRepository;
-	
+
 	@Autowired
 	private ItemRepository itemRepository;
 
 	@Autowired
 	private ToppingRepository toppingRepository;
+	
+	@Autowired
```
Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:46]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:46]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'member def modifier' has incorrect indentation level 8, expected level should be 2.

File: src/main/java/com/example/service/SendOrderService.java [hunk 6]
```diff
@@ -33,41 +36,45 @@
...
+	private JavaMailSender mailSender;
```
Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:47]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/com/example/service/SendOrderService.java [hunk 7]
```diff
@@ -33,41 +36,45 @@
...
 
 	/**
 	 * 注文状況を支払い方法に応じて更新します.
 	 * 
 	 * @param form フォーム
 	 * @return 注文情報
 	 */
+	@Async
```
Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:55]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:55]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def modifier' has incorrect indentation level 8, expected level should be 2.

File: src/main/java/com/example/service/SendOrderService.java [hunk 8]
```diff
@@ -33,41 +36,45 @@
...
 	public Order sendOrder(SendOrderForm form) {
 		Order order = orderRepository.load(form.getId());
-		
+
 		BeanUtils.copyProperties(form, order);
-		String delevery = form.getOrderDate()+ " " + form.getDeliveryTime();
+		String delevery = form.getOrderDate() + " " + form.getDeliveryTime();
```
Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:60]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:60]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/com/example/service/SendOrderService.java [hunk 9]
```diff
@@ -33,41 +36,45 @@
...
-        try {
-            parsedDate = dateFormat.parse(delevery);
-        } catch (ParseException e) {
-            e.printStackTrace();
-        }
-        Timestamp timestamp = new java.sql.Timestamp(parsedDate.getTime());
-        
-        order.setOrderDate(Date.valueOf(form.getOrderDate()));
-        order.setDeliveryTime(timestamp);
+		java.util.Date parsedDate = null;
```
Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:62]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:62]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/com/example/service/SendOrderService.java [hunk 10]
```diff
@@ -33,41 +36,45 @@
...
+		try {
```
Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:63]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:63]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'try' has incorrect indentation level 16, expected level should be 4.

File: src/main/java/com/example/service/SendOrderService.java [hunk 11]
```diff
@@ -33,41 +36,45 @@
...
+			parsedDate = dateFormat.parse(delevery);
```
Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:64]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:64]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'try' child has incorrect indentation level 24, expected level should be 6.

File: src/main/java/com/example/service/SendOrderService.java [hunk 12]
```diff
@@ -33,41 +36,45 @@
...
+		} catch (ParseException e) {
```
Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:65]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:65]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'try rcurly' has incorrect indentation level 16, expected level should be 4.

File: src/main/java/com/example/service/SendOrderService.java [hunk 13]
```diff
@@ -33,41 +36,45 @@
...
+			e.printStackTrace();
```
Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:66]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:66]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'catch' child has incorrect indentation level 24, expected level should be 6.

File: src/main/java/com/example/service/SendOrderService.java [hunk 14]
```diff
@@ -33,41 +36,45 @@
...
+		}
```
Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:67]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:67]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'catch rcurly' has incorrect indentation level 16, expected level should be 4.

File: src/main/java/com/example/service/SendOrderService.java [hunk 15]
```diff
@@ -33,41 +36,45 @@
...
+		Timestamp timestamp = new java.sql.Timestamp(parsedDate.getTime());
```
Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:68]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:68]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/com/example/service/SendOrderService.java [hunk 16]
```diff
@@ -33,41 +36,45 @@
...
+
+		order.setOrderDate(Date.valueOf(form.getOrderDate()));
```
Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:70]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:70]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/com/example/service/SendOrderService.java [hunk 17]
```diff
@@ -33,41 +36,45 @@
...
+		order.setDeliveryTime(timestamp);
```
Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:71]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:71]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/com/example/service/SendOrderService.java [hunk 18]
```diff
@@ -80,9 +87,21 @@ public Order sendOrder(SendOrderForm form) {
 			orderItem.setItem(item);
 		}
 		order.setOrderItemList(cartContent);
-		
+
 		order.setTotalPrice(order.getCalcPrice());
 		orderRepository.update(order);
+		SimpleMailMessage msg = new SimpleMailMessage();
```
Comment (tkeshun) [inline src/main/java/com/example/service/SendOrderService.java:93]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

**Review by tkeshun — COMMENTED** [review]
Remaining comments which cannot be posted as a review comment to avoid GitHub Rate Limit

<details>
<summary>checkstyle</summary>

[src/main/java/com/example/service/SendOrderService.java|93 col 17|](http://github.com/ecsite-project/ec-202304c/blob/3e945337774fe6d7d10a1b81a484c8a183e510ba/src/main/java/com/example/service/SendOrderService.java#L93) 'method def' child has incorrect indentation level 16, expected level should be 4.
[src/main/java/com/example/service/SendOrderService.java|94 col 1|](http://github.com/ecsite-project/ec-202304c/blob/3e945337774fe6d7d10a1b81a484c8a183e510ba/src/main/java/com/example/service/SendOrderService.java#L94) Line contains a tab character.
[src/main/java/com/example/service/SendOrderService.java|94 col 17|](http://github.com/ecsite-project/ec-202304c/blob/3e945337774fe6d7d10a1b81a484c8a183e510ba/src/main/java/com/example/service/SendOrderService.java#L94) 'method def' child has incorrect indentation level 16, expected level should be 4.
[src/main/java/com/example/service/SendOrderService.java|95 col 1|](http://github.com/ecsite-project/ec-202304c/blob/3e945337774fe6d7d10a1b81a484c8a183e510ba/src/main/java/com/example/service/SendOrderService.java#L95) Line contains a tab character.
[src/main/java/com/example/service/SendOrderService.java|95 col 17|](http://github.com/ecsite-project/ec-202304c/blob/3e945337774fe6d7d10a1b81a484c8a183e510ba/src/main/java/com/example/service/SendOrderService.java#L95) 'method def' child has incorrect indentation level 16, expected level should be 4.
[src/main/java/com/example/service/SendOrderService.java|96 col 1|](http://github.com/ecsite-project/ec-202304c/blob/3e945337774fe6d7d10a1b81a484c8a183e510ba/src/main/java/com/example/service/SendOrderService.java#L96) Line contains a tab character.
[src/main/java/com/example/service/SendOrderService.java|96 col 17|](http://github.com/ecsite-project/ec-202304c/blob/3e945337774fe6d7d10a1b81a484c8a183e510ba/src/main/java/com/example/service/SendOrderService.java#L96) 'method def' child has incorrect indentation level 16, expected level should be 4.
[src/main/java/com/example/service/SendOrderService.java|97 col 1|](http://github.com/ecsite-project/ec-202304c/blob/3e945337774fe6d7d10a1b81a484c8a183e510ba/src/main/java/com/example/service/SendOrderService.java#L97) Line contains a tab character.
[src/main/java/com/example/service/SendOrderService.java|97 col 17|](http://github.com/ecsite-project/ec-202304c/blob/3e945337774fe6d7d10a1b81a484c8a183e510ba/src/main/java/com/example/service/SendOrderService.java#L97) 'method def' child has incorrect indentation level 16, expected level should be 4.
[src/main/java/com/example/service/SendOrderService.java|99 col 1|](http://github.com/ecsite-project/ec-202304c/blob/3e945337774fe6d7d10a1b81a484c8a183e510ba/src/main/java/com/example/service/SendOrderService.java#L99) Line contains a tab character.
[src/main/java/com/example/service/SendOrderService.java|99 col 17|](http://github.com/ecsite-project/ec-202304c/blob/3e945337774fe6d7d10a1b81a484c8a183e510ba/src/main/java/com/example/service/SendOrderService.java#L99) 'try' has incorrect indentation level 16, expected level should be 4.
[src/main/java/com/example/service/SendOrderService.java|100 col 1|](http://github.com/ecsite-project/ec-202304c/blob/3e945337774fe6d7d10a1b81a484c8a183e510ba/src/main/java/com/example/service/SendOrderService.java#L100) Line contains a tab character.
[src/main/java/com/example/service/SendOrderService.java|100 col 25|](http://github.com/ecsite-project/ec-202304c/blob/3e945337774fe6d7d10a1b81a484c8a183e510ba/src/main/java/com/example/service/SendOrderService.java#L100) 'try' child has incorrect indentation level 24, expected level should be 6.
[src/main/java/com/example/service/SendOrderService.java|101 col 1|](http://github.com/ecsite-project/ec-202304c/blob/3e945337774fe6d7d10a1b81a484c8a183e510ba/src/main/java/com/example/service/SendOrderService.java#L101) Line contains a tab character.
[src/main/java/com/example/service/SendOrderService.java|101 col 17|](http://github.com/ecsite-project/ec-202304c/blob/3e945337774fe6d7d10a1b81a484c8a183e510ba/src/main/java/com/example/service/SendOrderService.java#L101) 'try rcurly' has incorrect indentation level 16, expected level should be 4.
[src/main/java/com/example/service/SendOrderService.java|102 col 1|](http://github.com/ecsite-project/ec-202304c/blob/3e945337774fe6d7d10a1b81a484c8a183e510ba/src/main/java/com/example/service/SendOrderService.java#L102) Line contains a tab character.
[src/main/java/com/example/service/SendOrderService.java|102 col 25|](http://github.com/ecsite-project/ec-202304c/blob/3e945337774fe6d7d10a1b81a484c8a183e510ba/src/main/java/com/example/service/SendOrderService.java#L102) 'catch' child has incorrect indentation level 24, expected level should be 6.
[src/main/java/com/example/service/SendOrderService.java|103 col 1|](http://github.com/ecsite-project/ec-202304c/blob/3e945337774fe6d7d10a1b81a484c8a183e510ba/src/main/java/com/example/service/SendOrderService.java#L103) Line contains a tab character.
[src/main/java/com/example/service/SendOrderService.java|103 col 17|](http://github.com/ecsite-project/ec-202304c/blob/3e945337774fe6d7d10a1b81a484c8a183e510ba/src/main/java/com/example/service/SendOrderService.java#L103) 'catch rcurly' has incorrect indentation level 16, expected level should be 4.
</details>

**Review by yousuke-murayama — APPROVED** [review]
問題ないと思います。

**Pull request merged**